src/Model.jl

```diff
@@ -65,32 +65,64 @@ end
 _is_well_conditioned(A; tol=1e5) = (cond(A) <= tol)
 
 """
-    Model(lattice, atoms, positions; n_electrons, magnetic_moments, terms, temperature,
-          smearing, spin_polarization, symmetries)
+    Model(lattice, atoms, positions; model_name, εF, n_electrons,
+                disable_electrostatics_check, magnetic_moments, terms,
+                temperature, smearing, spin_polarization, symmetries)
 
 Creates the physical specification of a model (without any discretization information).
 
-`n_electrons` is taken from `atoms` if not specified.
+The `Model` represents the model system through its geometry (lattice, atomic positions),
+electronic structure (number of electrons, spin polarization), and physical terms
+(kinetic energy, external potentials, Hartree, exchange-correlation, etc.). 
+It does not include discretization parameters; those are set separately when constructing
+a [`PlaneWaveBasis`](@ref) from a `Model`.
 
-`spin_polarization` is :none by default (paired electrons)
-unless any of the elements has a non-zero initial magnetic moment.
-In this case the spin_polarization will be :collinear.
+Arguments
+- `lattice::AbstractMatrix`: Lattice vectors given column-wise (atomic units).
+- `atoms::Vector{<:Element}`: Atomic species (may include pseudopotential metadata).
+- `positions::Vector{Vec3}`: Atomic positions in fractional (reduced) coordinates.
 
-`magnetic_moments` is only used to determine the symmetry and the
-`spin_polarization`; it is not stored inside the datastructure.
-
-`smearing` is Fermi-Dirac if `temperature` is non-zero, none otherwise
-
-The `symmetries` kwarg allows (a) to pass `true` / `false` to enable / disable
-the automatic determination of lattice symmetries or (b) to pass an explicit list
-of symmetry operations to use for lowering the computational effort.
-The default behaviour is equal to `true`, namely that the code checks the
-specified model in form of the Hamiltonian `terms`, `lattice`, `atoms` and `magnetic_moments`
-parameters and from these automatically determines a set of symmetries it can safely use.
-If you want to pass custom symmetry operations (e.g. a reduced or extended set) use the
-`symmetry_operations` function. Notice that this may lead to wrong results if e.g. the
-external potential breaks some of the passed symmetries. Use `false` to turn off
-symmetries completely.
+## Keyword arguments
```

Comment:
I would put the expert arguments `model_name`, `εF`, `disable_electrostatics_check` last and annotate these are expert.

Reply:
Some of this information is fairly stable, I agree. But I would like to keep duplication to a minimum. Is there any way with https://github.com/JuliaDocs/DocumenterTools.jl to get the type and defaults automatically from the function definition ?

Reply:
Ok I put it on the hackathon tasks list for next time